src/inmanta/data/dataview.py

```diff
@@ -1271,27 +1272,39 @@ def __init__(
             prelude=None,
         )
         self._prelude_query_builder = prelude_query_builder
+        self._prelude_query_builder_extra = prelude_query_builder_extra
 
     @property
     def offset(self) -> int:
         """The current offset of the values to be used for filter statements"""
-        return len(self.values) + len(self._prelude_query_builder.values) + 1
+        extra_values_count = sum(len(builder.values) for builder in self._prelude_query_builder_extra)
+        return len(self.values) + len(self._prelude_query_builder.values) + extra_values_count + 1
 
     def build(self) -> tuple[str, list[object]]:
         prelude_query, prelude_values = self._prelude_query_builder.build()
-        prelude_query_in_with_block = f"WITH prelude AS ({prelude_query})"
-        delegate: SimpleQueryBuilder = SimpleQueryBuilder(
+        prelude_section = f"prelude AS ({prelude_query})"
+        extra_prelude_values = []
+        extra_prelude_queries = []
+
+        for idx, extra_builder in enumerate(self._prelude_query_builder_extra):
+            extra_query, extra_values = extra_builder.build()
```

Comment:
How do you make sure that the parameter index of this query doesn't overlap with the previous prelude?

Reply:
Indeed this is an issue, I don't know yet how to fix it. I encountered it on my branch on lsm: both my prelude use the same parameter but the query is now expecting 2 different parameters. I'll have to think about this.